trewrap/src/main/java/org/codice/imaging/nitf/trewrap/SECURA.java

```diff
@@ -0,0 +1,204 @@
+            NitfInputStreamReader nitfReader = new NitfInputStreamReader(stream);
+            nitfReader.setFileType(getNitfVersion());
+            securityMetadata = parser.parseSecurityMetadata(nitfReader);
+        } else {
+            securityMetadata = null;
+                throw new NitfFormatException("Could not parse Security Fields");
+        }
+        return securityMetadata;
+    }
+
+    /**
+     * Get Security Standard field value.
+     *
+     * The security standard used to populate the SECURITY field.
+     *
+     * @return the security standard string
+     * @throws NitfFormatException if there is a parsing issue.
+     */
+    public final String getSecurityStandard() throws NitfFormatException {
+        return getValueAsTrimmedString("SECSTD");
+    }
+
+    /**
+     * Get the SECURITY field compression.
+     *
+     * This field identifies the compression used, if any, of the SECURITY field.
+     *
+     * @return the SECURITY field compression string
+     * @throws NitfFormatException if there is a parsing issue.
+     */
+    public final String getSecurityFieldCompression() throws NitfFormatException {
+        return getValueAsTrimmedString("SECCOMP");
+    }
+
+    /**
+     * Get the length of the SECURITY field.
+     *
+     * This field holds the length of the SECURITY field in bytes.
+     *
+     * @return SECURITY field length
+     * @throws NitfFormatException if there is a parsing issue on the NITF side
+     */
+    public final int getSecurityLength() throws NitfFormatException {
+        return getValueAsInteger("SECLEN");
+    }
+
+    /**
+     * Get the Security field value.
+     *
+     * The actual security data, as encoded using the security standard specified in the
+     * SECSTD field and compressed as specified by the SECCOMP field.
+     *
+     * @return the Security data
+     * @throws NitfFormatException if there is a parsing issue.
+     */
+    public final byte[] getSecurity() throws NitfFormatException {
+        return getFieldValue("SECURITY").getBytes(StandardCharsets.ISO_8859_1);
+    }
+
+    /**
+     * Get the Security field value.
+     *
+     * The actual security data, as encoded using the security standard specified in the
+     * SECSTD field and compressed as specified by the SECCOMP field.
+     *
+     * @return the uncompressed Security data
+     * @throws NitfFormatException if there is a parsing issue.
+     */
+    public final String getSecurityUncompressed() throws NitfFormatException {
+        if (getSecurityFieldCompression().equals("GZIP")) {
+            byte[] security = getSecurity();
+
+            StringBuilder sb = new StringBuilder();
+            try (ByteArrayInputStream bis = new ByteArrayInputStream(security);
+            GZIPInputStream gis = new GZIPInputStream(bis);
+            BufferedReader br = new BufferedReader(new InputStreamReader(gis, StandardCharsets.ISO_8859_1))) {
```

Comment:
Are you sure about the encoding here? There were some UTF-8 references earlier.

Reply:
Yes, the UTF-8 buffer in the test, is to convert the java string into ISO-8859-1 bytes since some UTF-8 characters are represented by two bytes.  The parser reads the bytes and converts them into ISO-8859-1 strings, and returns strings made from ISO-8859-1.  The compressed information is binary so w/o specifying ISO-8859-1 it can have extra bytes from the UTF-8 encoding.  

I can change the thrown error test to concatenate the gzipped security string and it will throw an error when fed into the tre parser as a string.

Reply:
I went back and looked for the UTF-8 references you spoke of and I found some places that should have been ISO-8859-1.  I updated the code.

Reply:
I'm unconvinced. The text in STDI-0002 Table AI-1 for SECURITY says (if the SECSTD is ARH.XML) that the uncompressed value is UTF-8. So the bytes coming out of the decompression are UTF-8 encoded.